### PR TITLE
Updated Calibration and OTA Multi-User Downlink

### PR DESCRIPTION
**Pull request by rdoosty**
- updates to calibration procesure which includes initial calibration and online calibration
- online calibration updates in ZF_freq_orthogonal function

### CODE REVIEW

**Review by asedlmayr — COMMENTED** [review]
Tested with the flows I have experience with.  Don't see any obvious regressions.  The tests still have some memory issues in debug mode but I have cleaned that up in another branch.

**Review by rdoosty — COMMENTED** [review]
Thanks Andrew for your review. See my changes and responses below.

File: src/agora/dofft.cpp [hunk 1]
```diff
@@ -378,16 +408,18 @@ Event_data DoIFFT::launch(size_t tag)
     // IFFT scaled results by OFDM_CA_NUM, we scale down IFFT results
     // during data type coversion
     simd_convert_float_to_short(ifft_out_ptr, socket_ptr, cfg->OFDM_CA_NUM,
-        cfg->CP_LEN, cfg->OFDM_CA_NUM);
+        cfg->CP_LEN, cfg->OFDM_CA_NUM / std::sqrt(cfg->BF_ANT_NUM * 1.f));
```
Comment (jianding17) [inline src/agora/dofft.cpp:411]: We should avoid doing division here due to performance issue. Maybe we can compute `cfg->OFDM_CA_NUM / std::sqrt(cfg->BF_ANT_NUM * 1.f)` in the constructor of doIFFT?

Comment (rdoosty) [reply src/agora/dofft.cpp]: Done!

**Pull request merged**